lib/spack/spack/cmd/diy.py

```diff
@@ -62,6 +68,10 @@ def diy(self, args):
     if not args.spec:
         tty.die("spack diy requires a package spec argument.")
 
+    if args.jobs is not None:
```

Comment:
Such factoring has already been done (extensively) in #2664.  Please base your changes off of that PR.